tests/test_io_elementwise.py

```diff
@@ -156,30 +213,113 @@ def test_io_spec_cupy(store, value, encoding_type, as_dask):
-    import dask.array as da
+def test_dask_write_sparse(sparse_format, store):
+    x_sparse_store = create_sparse_store(sparse_format, store)
+    X_from_disk = read_elem(x_sparse_store["X"])
+    X_dask_from_disk = read_elem(x_sparse_store["X_dask"])
 
-    X = sparse.random(
-        1000,
-        1000,
-        format=sparse_format,
-        density=0.01,
-        random_state=np.random.default_rng(),
-    )
-    X_dask = da.from_array(X, chunks=(100, 100))
+    assert_equal(X_from_disk, X_dask_from_disk)
+    assert_equal(dict(x_sparse_store["X"].attrs), dict(x_sparse_store["X_dask"].attrs))
 
-    write_elem(store, "X", X)
-    write_elem(store, "X_dask", X_dask)
+    assert x_sparse_store["X_dask/indptr"].dtype == np.int64
+    assert x_sparse_store["X_dask/indices"].dtype == np.int64
 
-    X_from_disk = read_elem(store["X"])
-    X_dask_from_disk = read_elem(store["X_dask"])
+
+def test_read_lazy_2d_dask(sparse_format, store):
+    arr_store = create_sparse_store(sparse_format, store)
+    X_dask_from_disk = read_elem_as_dask(arr_store["X"])
+    X_from_disk = read_elem(arr_store["X"])
 
     assert_equal(X_from_disk, X_dask_from_disk)
-    assert_equal(dict(store["X"].attrs), dict(store["X_dask"].attrs))
+    random_int_indices = np.random.randint(0, SIZE, (SIZE // 10,))
+    random_int_indices.sort()
+    random_bool_mask = np.random.randn(SIZE) > 0
+    index_slice = slice(0, SIZE // 10)
+    for index in [random_int_indices, index_slice, random_bool_mask]:
+        assert_equal(X_from_disk[index, :], X_dask_from_disk[index, :])
+        assert_equal(X_from_disk[:, index], X_dask_from_disk[:, index])
+
+    assert arr_store["X_dask/indptr"].dtype == np.int64
+    assert arr_store["X_dask/indices"].dtype == np.int64
+
+
+@pytest.mark.parametrize(
+    ("n_dims", "chunks"),
+    [
+        (1, (100,)),
+        (1, (400,)),
+        (2, (100, 100)),
+        (2, (400, 400)),
+        (2, (200, 400)),
+        (1, None),
+        (2, None),
+    ],
+)
+def test_read_lazy_subsets_nd_dask(store, n_dims, chunks):
+    arr_store = create_dense_store(store, n_dims)
+    X_dask_from_disk = read_elem_as_dask(arr_store["X"], chunks=chunks)
+    X_from_disk = read_elem(arr_store["X"])
+    assert_equal(X_from_disk, X_dask_from_disk)
+
+    random_int_indices = np.random.randint(0, SIZE, (SIZE // 10,))
+    random_int_indices.sort()
+    random_bool_mask = np.random.randn(SIZE) > 0
+    index_slice = slice(0, SIZE // 10)
+    for index in [random_int_indices, index_slice, random_bool_mask]:
+        assert_equal(X_from_disk[index], X_dask_from_disk[index])
+
+
+def test_read_lazy_h5_cluster(sparse_format, tmp_path):
+    import dask.distributed as dd
+
+    file = h5py.File(tmp_path / "test.h5", "w")
+    store = file["/"]
+    arr_store = create_sparse_store(sparse_format, store)
+    X_dask_from_disk = read_elem_as_dask(arr_store["X"])
+    X_from_disk = read_elem(arr_store["X"])
+    file.close()
```

Comment:
```suggestion
    with h5py.File(tmp_path / "test.h5", "w") as file:
        store = file["/"]
        arr_store = create_sparse_store(sparse_format, store)
        X_dask_from_disk = read_elem_as_dask(arr_store["X"])
        X_from_disk = read_elem(arr_store["X"])
```

Should be the same, and more common way to do this